extensions/notebook/src/protocol/notebookUriHandler.ts

```diff
@@ -112,6 +128,17 @@ export class NotebookUriHandler implements vscode.UriHandler {
 		});
 	}
 
+	private extractNotebookNameFromUrl(url: string): string | undefined {
+		let escapedUrl = decodeURI(url);
+		let nameMatches = escapedUrl.match(/(.+\/)+(.+\.ipynb)/);
+
+		if (nameMatches) {
+			return nameMatches[2];
```

Comment:
I'm not sure this regex is correct - there's only going to be 2 capturing groups unless I'm missing something so index 2 is always going to be invalid. And what's the format of the URL you're expecting?

Reply:
We also have built in stuff for decoding URIs into their proper components. See src\sql\workbench\contrib\commandLine\electron-browser\commandLine.ts

We break down URIs here

Reply:
Agreed. This is only in the extension because of legacy reasons. I would love to see this move into core.

Reply:
The `match` method returns the original string at index 0, and matched groups thereafter. Please view sample link here: [http://regex101.com/r/UV8YWx/4/](http://regex101.com/r/UV8YWx/4/). 

Got it, will look into integrating this in Core. @chlafreniere What do you want me to do with the existing handler? Should I leave it in favor of not breaking other people using these links?

Reply:
I think as long as the same URIs still work from a user perspective, then it's ok. We can totally chat about this, though; maybe we can setup a quick chat with @anthonydresser?

I've been wanting to get some of the notebook commands that aren't Jupyter-specific out of the extension, since those are just there for legacy reasons (since our initial notebook implementation lived only in the extension); notebook.command.new and notebook.command.open are 2 of my candidates for moving 😄.

I would love it if we could have generic commands that we used here instead of notebook-specific ones; a generic file open and new file URI.